pywbemtools/pywbemcli/_connection_repository.py

```diff
@@ -255,76 +355,83 @@ def _load_connections_file(self):
-                                     (self._connections_file, ve))
+                        raise ConnectionsFileLoadError(
+                            self._connections_file,
+                            'Invalid object type server name: "{0}". '
+                            'Item: "{1}"'.format(name, te))
+
+                except (TypeError, yaml.YAMLError) as exc:
+                    raise ConnectionsFileLoadError(
+                        self._connections_file,
+                        'Invalid YAML. Exception: {0}'.format(exc))
+
         else:
-            raise ConnectionsFileNotFoundError(
-                "Connections file does not exist: {0}".
-                format(self.connections_file))
+            raise ConnectionsFileNotFoundError(self.connections_file)
 
-    def add(self, name, svr_definition):
+    def add(self, svr_definition):
         """
         Add a new connection to the connections repository or replace an
         existing connection.  Users of this method should check before add if
         they do not want to replace an existing entry.
 
         Parameter:
 
-          svr_definition (:class:`ServerDefinition`):
-            An instance of ServerDefinition that contains the data that will be
-            added to the connections file.
+          svr_definition (:class:`PywbemServer`):
+            An instance of PywbemServer that contains the data that will be
+            added to the connections file. The server definition must contain
+            a name attribute.
 
         Raises:
-            click.Abort if the load of the connections file returns an
-            exception
+            ConnectionFileLoadError: if the file is invalid
+            ConnectionFileWriteError: if it cannot write the modified file.
         """
+        assert isinstance(svr_definition.name, six.string_types)
 
-        # If the file does exists get it. Otherwise the write will create it
-        # TODO: we process errors here differently than in read mode because
-        # de use click and abort. Should we do this in common.
-        # Also, why not make abort_click a common function with message.
+        # If the file exists load it. Otherwise the write will create it
         if os.path.isfile(self._connections_file):
-            try:
-                self._load_connections_file()
-            # All other errors cases abort.
-            except Exception as e:
-                click.echo("Fatal error loading YAML file {0}. "
-                           "Exception: {1}: {2}".
-                           format(self.connections_file, e.__class__.__name__,
-                                  e),
-                           err=True)
-                raise click.Abort()
-
-        self._pywbemcli_servers[name] = svr_definition
+            self._load_connections_file()
+
+        self._pywbemcli_servers[svr_definition.name] = svr_definition
         self._write_file()
```

Comment:
A nit, but can we use consistent names for `_load_connections_file()` and `_write_file()`?